plugins/notifications-backend/src/service/router.ts

```diff
@@ -23,7 +25,14 @@ import { RouterOptions } from './types';
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const { logger, dbConfig, catalogClient } = options;
+  const { logger, database, discovery, config } = options;
+
+  // workaround for creating the database when client is not sqlite
+  const dbClientWorkaround = await database.getClient();
```

Comment:
this should not be here. the db creation whether required or not (sqlite does not reuqire it) is not part of the plugin. is it a limitation due to dynamic plugin?

Reply:
yes, to unify the flows